File: csharp/src/CQRSService.Client/Queries/GetClientQueryHandler.cs
using CQRSService.Client.Domain.Queries;
using CQRSService.Client.Storage;
using CQRSService.Domain.Queries;
using System.Threading;
using System.Threading.Tasks;

namespace CQRSService.Client.Queries
{
    public class GetClientQueryHandler : IQueryHandler<GetClientQuery, Client>
    {
        private readonly IStorage<Client> _clientStorage;

        public GetClientQueryHandler(IStorage<Client> clientStorage) => _clientStorage = clientStorage;
        public Task<Client> Handle(GetClientQuery query, CancellationToken cancellationToken)
            =>  _clientStorage.GetOne(query.Id);
    }
}

File: golang/client/queries/getClientQueryHandler.go
package client

import (
	client "github.com/houstondapaz/cqrs-govscsharp/golang/client/domain"
	queries "github.com/houstondapaz/cqrs-govscsharp/golang/domain/queries"
)

type GetClientQueryHandler struct {
	queries.QueryHandler
	storage client.ClientStorager
}

func NewGetClientQueryHandler(storage client.ClientStorager) GetClientQueryHandler {
	return GetClientQueryHandler{
		storage: storage,
	}
}

func (qh GetClientQueryHandler) QueryName() string {
	return "GetClientQuery"
}

func (qh GetClientQueryHandler) Handle(query queries.Querier) (interface{}, error) {
	getClientQuery := query.(client.GetClientQuery)
	client, err := qh.storage.GetOne(getClientQuery.ClientId())

	if err != nil {
		return nil, err
	}

	return client, nil
}
